include/bluetooth/audio/audio.h

```diff
@@ -197,6 +197,37 @@ struct bt_codec_data {
 					  BT_CODEC_META_CONTEXT_RINGTONE | \
 					  BT_CODEC_META_CONTEXT_TV)
 
+enum bt_audio_location {
+	BT_AUDIO_LOCATION_FRONT_LEFT = BIT(0),
+	BT_AUDIO_LOCATION_FRONT_RIGHT = BIT(1),
+	BT_AUDIO_LOCATION_FRONT_CENTER = BIT(2),
+	BT_AUDIO_LOCATION_LOW_FREQ_EFFECTS_1 = BIT(3),
+	BT_AUDIO_LOCATION_BACK_LEFT = BIT(4),
+	BT_AUDIO_LOCATION_BACK_RIGHT = BIT(5),
+	BT_AUDIO_LOCATION_FRONT_LEFT_OF_CENTER = BIT(6),
+	BT_AUDIO_LOCATION_FRONT_RIGHT_OF_CENTER = BIT(7),
+	BT_AUDIO_LOCATION_BACK_CENTER = BIT(8),
+	BT_AUDIO_LOCATION_LOW_FREQ_EFFECTS_2 = BIT(9),
+	BT_AUDIO_LOCATION_SIDE_LEFT = BIT(10),
+	BT_AUDIO_LOCATION_SIDE_RIGHT = BIT(11),
+	BT_AUDIO_LOCATION_TOP_FRONT_LEFT = BIT(12),
+	BT_AUDIO_LOCATION_TOP_FRONT_RIGHT = BIT(13),
+	BT_AUDIO_LOCATION_TOP_FRONT_CENTER = BIT(14),
+	BT_AUDIO_LOCATION_TOP_CENTER = BIT(15),
+	BT_AUDIO_LOCATION_TOP_BACK_LEFT = BIT(16),
+	BT_AUDIO_LOCATION_TOP_BECK_RIGHT = BIT(17),
+	BT_AUDIO_LOCATION_TOP_SIDE_LEFT = BIT(18),
+	BT_AUDIO_LOCATION_TOP_SIDE_RIGHT = BIT(19),
+	BT_AUDIO_LOCATION_TOP_BACK_CENTER = BIT(20),
+	BT_AUDIO_LOCATION_BOTTOM_FRONT_CENTER = BIT(21),
+	BT_AUDIO_LOCATION_BOTTOM_FRONT_LEFT = BIT(22),
+	BT_AUDIO_LOCATION_BOTTOM_FRONT_RIGHT = BIT(23),
+	BT_AUDIO_LOCATION_FRONT_LEFT_WIDE = BIT(24),
+	BT_AUDIO_LOCATION_FRONT_RIGHT_WIDE = BIT(25),
+	BT_AUDIO_LOCATION_LEFT_SURROUND = BIT(26),
+	BT_AUDIO_LOCATION_RIGHT_SURROUND = BIT(27),
+};
+
```

Comment:
Even though it makes sense to group these together, I do not think using an enum is a good way to do it. 
Enums contains a finite number of possible values - hence you would need to suppress compiler warnings to be able to OR multiple values together - or you would need to typecast the result back to the enum type - which could result in errors if the compiler have made assumptions about the finite number of values (for example a switch statement do not need a default case when doing switch on an enum type and all possible cases are handled). I do not know if newer c-versions have support for this though?
I have seen others define a type and then typecast to this in the define:

`typedef uint32_t bt_audio_loc_t`
`#define BT_AUDIO_LOCATION_FRONT_LEFT ((bt_audio_loc_t) BIT(0))`

then you can use bt_audio_loc_t rather than uint32_t for function parameters that expect audio location(s). And you can freely OR multiple together as the result would still be uint32_t.

To group them together in the API documentation use the `@name Audio Locations @{ ... @} `grouping feature

Reply:
I hear your concerns and see what issues this may cause. 

The reasoning behind this approach is:

1. (named) `enum`s provide an easy way for an API to express the "group" of values expected, without relying on doxygen-specific grouping
2. https://www.kernel.org/doc/html/v4.10/process/coding-style.html#macros-enums-and-rtl (which we follow) express that `enum`s are preferred for groups of constants
3. This is the approach (although without the name) used by e.g. `BT_LE_ADV_OPT_`, and other option `enum`s in `bluetooth.h`. 

If we see compiler issues, then this is indeed something we need to fix. If the issue is caused by using the `enum` type in the APIs, then we can replace the `enum` type with `uint32_t` (similar to how the `BT_LE_ADV_OPT_` are used), and simply add a reference to the `enum` in the function description. 

If the issue is caused by just using the `enum` keyword, then this problem exist in many places in the code base, and we need a general fix as well as an exception to the Linux kernel code guideline. 

Do you have any references to the compiler issues (warnings) you are experiencing that I can read?

Reply:
If this is an accepted way to do it in Zephyr, I'm OK with it.
I do not have a reference, but in my early days I was told to refactor a full FM radio API I made due to this issue (not a GCC based compiler) - hence I always try to avoid using enums when I specify bit-values. 
I had a look at the different enum related warnings that GCC support, and non of them covers this issue, hence I think we can go ahead using enums in this case.

Reply:
Otherwise refactoring to a group of `#define`s can be done without breaking the API if we just change the API from `enum bt_audio_location` to `uint32_t bt_audio_location`.